src/conversion/model-selection-to-view-converters.js

```diff
@@ -145,27 +145,54 @@ export function convertCollapsedSelection() {
  */
 export function convertSelectionAttribute( elementCreator ) {
 	return ( evt, data, consumable, conversionApi ) => {
-		const selection = data.selection;
+		const viewElement = elementCreator instanceof ViewElement ?
+			elementCreator.clone( true ) :
+			elementCreator( data.value, data, data.selection, consumable, conversionApi );
 
-		if ( !selection.isCollapsed ) {
-			return;
-		}
+		convertSelectionPosition( data.selection, conversionApi.viewSelection, viewElement, consumable, 'selectionAttribute:' + data.key );
+	};
+}
 
-		if ( !consumable.consume( selection, 'selectionAttribute:' + data.key ) ) {
-			return;
-		}
+/**
+ * Performs similar conversion as {@link ~convertSelectionAttribute}, but depends on a marker name of a marker in which
+ * collapsed selection is placed.
+ *
+ *		modelDispatcher.on( 'selectionMarker:searchResult', wrapRange( new ViewAttributeElement( 'span', { class: 'searchResult' } ) ) );
+ *
+ * **Note:** You can use the same `elementCreator` function for this converter factory
+ * and {@link module:engine/conversion/model-to-view-converters~wrapRange}.
+ *
+ * @see {~convertSelectionAttribute}
+ * @param {module:engine/view/attributeelement~AttributeElement|Function} elementCreator View element,
+ * or function returning a view element, which will be used for wrapping.
+ * @returns {Function} Selection converter.
+ */
+export function convertSelectionMarker( elementCreator ) {
+	return ( evt, data, consumable, conversionApi ) => {
+		const viewElement = elementCreator instanceof ViewElement ?
+			elementCreator.clone( true ) :
+			elementCreator( data, consumable, conversionApi );
 
-		let viewPosition = conversionApi.viewSelection.getFirstPosition();
-		conversionApi.viewSelection.removeAllRanges();
+		convertSelectionPosition( data.selection, conversionApi.viewSelection, viewElement, consumable, 'selectionMarker:' + data.name );
+	};
+}
 
-		const viewElement = elementCreator instanceof ViewElement ?
-				elementCreator.clone( true ) :
-				elementCreator( data.value, data, selection, consumable, conversionApi );
+// Helper function for `convertSelectionAttribute` and `convertSelectionMarker`, which perform similar task.
+function convertSelectionPosition( modelSelection, viewSelection, viewElement, consumable, consumableName ) {
```

Comment:
Does not method convert "full" selection (with attribute and markers)? If not then the name is misleading. Maybe `wrapSelectionPosition`?